lib/tarantool_server.py

```diff
@@ -481,11 +481,19 @@ def seek_wait(self, msg, proc=None, name=None, deadline=None):
                         raise TarantoolStartError(name)
                 log_str = f.readline()
                 if not log_str:
-                    gevent.sleep(0.001)
+                    # We reached the end of the logfile.
                     f.seek(cur_pos, os.SEEK_SET)
+                    gevent.sleep(0.001)
                     continue
-                if re.findall(msg, log_str):
-                    return True
+                else:
+                    # if the whole line is read, check the pattern in the line.
+                    # Otherwise, set the cursor back to read the line again.
+                    if log_str.endswith('\n'):
+                        if re.findall(msg, log_str):
+                            return True
+                    else:
+                        f.seek(cur_pos, os.SEEK_SET)
```

Comment:
This way the file descriptor will always be readable, it is busy loop. Either collect readed bytes into a buffer (preferable) or add a short sleep.

Reply:
A sleep is added.